tests/unit/type/set.tcl

```diff
@@ -1092,6 +1092,11 @@ if {[lindex [r config get proto-max-bulk-len] 1] == 10000000000} {
         assert_equal 1 [write_big_bulk $str_length "bbb"]
         assert_equal [read_big_bulk {r spop myset} yes "aaa"] $str_length
     } {} {large-memory}
+
+    # restore defaults
+    r config set proto-max-bulk-len 536870912
```

Comment:
Makes me think the tests may need a more generic "push config" and "pop config" mechanism.

Reply:
we have `proc config_get_set`, but we don't use it as much, and also it doesn't really solve much of the problem (you need to remember to call it).

but anyway, the problem here was with external tests (unlike #11636), and we do have this in server.tcl
```tcl
    # restore overrides
    dict for {param val} $saved_config {
        r config set $param $val
    }
```
maybe it could be extended to roll back all configs, between external servers, and not just the ones set by the server infra...